tests/EStimLibrary.UnitTests/Extensions/SpatialModel/StringHierarchy/StringHierarchySpecTests.cs

```diff
@@ -0,0 +1,552 @@
+        new[] { "RegionA", "RegionB" }, new[] { "Modifier2", "Modifier1" }, true)]
+    [InlineData(new[] { "RegionA", "RegionB" }, new[] { "Modifier1", "Modifier2" },
+        new[] { "RegionA", "RegionB" }, new[] { "Modifier1" }, false)]
+    [InlineData(new[] { "RegionA" }, new[] { "Modifier1" },
+        new[] { "RegionA" }, new[] { "Modifier1" }, true)]
+    [InlineData(new[] { "RegionA", "RegionB" }, new[] { "Modifier1" },
+        new[] { "RegionA", "RegionC" }, new[] { "Modifier1" }, false)]
+    [InlineData(new[] { "RegionA" }, new[] { "Modifier1", "Modifier2" },
+        new[] { "RegionA" }, new[] { "Modifier2", "Modifier1" }, true)]
+    [InlineData(new[] { "RegionA", "RegionB" }, new[] { "Modifier1" },
+        new[] { "" }, new[] { "" }, false)]
+    public void Equals_ShouldReturnExpectedResult(
+        string[] regionSet1, string[] modifierSet1,
+    string[] regionSet2, string[] modifierSet2, bool expected)
+    {
+        var spec1 = new StringHierarchySpec(regionSet1, modifierSet1);
+        var spec2 = regionSet2 != null ? new StringHierarchySpec(regionSet2, modifierSet2) : null;
+        bool result = spec1.Equals(spec2);
+        Assert.Equal(expected, result);
+    }
+
+    /// <summary>
+    /// Test the GetHashCode method. 
+    /// </summary>
+    [Theory]
+    [InlineData(new[] { "RegionA", "RegionB" }, new[] { "Modifier1", "Modifier2" },
+        new[] { "RegionA", "RegionB" }, new[] { "Modifier2", "Modifier1" }, true)]
+    [InlineData(new[] { "RegionA" }, new[] { "Modifier1" },
+        new[] { "RegionA" }, new[] { "Modifier1" }, true)]
+    [InlineData(new[] { "RegionA", "RegionB" }, new[] { "Modifier1" },
+        new[] { "RegionA", "RegionC" }, new[] { "Modifier1" }, false)]
+    [InlineData(new[] { "RegionA", "RegionB" }, new[] { "Modifier1", "Modifier2" },
+        new[] { "RegionB" }, new[] { "Modifier1", "Modifier2" }, false)]
+    public void GetHashCode_ShouldReturnConsistentHashCodesForEqualObjects(
+        string[] regionSet1, string[] modifierSet1,
+        string[] regionSet2, string[] modifierSet2, bool expectEqualHashCodes)
+    {
+        var spec1 = new StringHierarchySpec(regionSet1, modifierSet1);
+        var spec2 = new StringHierarchySpec(regionSet2, modifierSet2);
+        int hash1 = spec1.GetHashCode();
+        int hash2 = spec2.GetHashCode();
+        if (expectEqualHashCodes)
+        {
+            Assert.Equal(hash1, hash2);
+        }
+        else
+        {
+            Assert.NotEqual(hash1, hash2);
+        }
+    }
+
+    /// <summary>
+    /// Test the ToString method. 
+    /// </summary>
+    [Theory]
+    [InlineData(new[] { "RegionA", "RegionB" }, new[] { "Modifier1" }, "RegionA, RegionB | Modifier1")]
+    [InlineData(new[] { "Main" }, new string[] { }, "Main")]
+    [InlineData(new[] { "X", "Y", "Z" }, new[] { "North", "South" }, "X, Y, Z | North, South")]
+    public void ToString_ShouldReturnFullSpecRepresentation(
```

Comment:
All internal representations should be stored as lowercase since they go through the Parse...() methods when the object is constructed. --> Why are these uppercase "expected" values succeeding? --> May be a bug somewhere that needs fixing